packages/web3torrent/src/pages/upload/Upload.test.tsx

```diff
@@ -0,0 +1,80 @@
+import Enzyme, {mount} from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+import {createMemoryHistory} from 'history';
+import React from 'react';
+import {act} from 'react-dom/test-utils';
+import {MemoryRouter as Router, RouteComponentProps} from 'react-router-dom';
+import {EmptyTorrent} from '../../constants';
+import {WebTorrentSeedInput} from '../../library/types';
+import {Status, Torrent} from '../../types';
+import {testSelector} from '../../utils/test-utils';
+import * as TorrentStatus from '../../utils/torrent-status-checker';
+import * as Web3TorrentClient from './../../clients/web3torrent-client';
+import Upload from './Upload';
+
+Enzyme.configure({adapter: new Adapter()});
+
+function setup() {
+  const history = createMemoryHistory({initialEntries: ['/upload']});
+  const props: RouteComponentProps = {
+    history,
+    location: history.location,
+    match: {
+      isExact: true,
+      params: {},
+      path: '/',
+      url: 'http://localhost/'
+    }
+  };
+
+  const torrentUpload = jest
+    .spyOn(Web3TorrentClient, 'upload')
+    .mockImplementation(_pD => Promise.resolve({...EmptyTorrent, status: Status.Seeding}));
+
+  const component = mount(
+    <Router>
+      <Upload {...props} />
+    </Router>
+  );
+
+  return {props, component, torrentUpload};
+}
+
+describe('<Upload />', () => {
+  let component: Enzyme.ReactWrapper;
+  let torrentUpload: jest.SpyInstance<Promise<Torrent>, [WebTorrentSeedInput]>;
+
+  beforeEach(() => {
+    const mock = setup();
+    component = mock.component;
+    torrentUpload = mock.torrentUpload;
+    jest.useFakeTimers();
+  });
+
+  it('should render an Upload button', () => {
```

Comment:
We should check for the rest of the UI as well, using the `exists()` helper, at the very least. For this page, as a minimum, a check for having a `type="file"` input would be nice.